autodispose/src/main/java/com/uber/autodispose/AutoDisposingObserverImpl.java

```diff
@@ -55,6 +55,8 @@
 
           @Override public void onComplete() {
             callMainSubscribeIfNecessary(d);
```

Comment:
When subscribing to a completed ScopeProvider, callMainSubscribeIfNecessary() might be called here before the main source is subscribed, which triggers the DoubleSubscriptoinsException.